gazebo/rendering/Light.cc

```diff
@@ -584,13 +585,19 @@ void Light::SetRange(const double _range)
 //////////////////////////////////////////////////
 void Light::SetCastShadows(const bool _cast)
 {
-  if (this->dataPtr->light->getType() == Ogre::Light::LT_DIRECTIONAL)
-  {
-    this->dataPtr->light->setCastShadows(_cast);
-  }
-  else
+  this->dataPtr->light->setCastShadows(_cast);
+
+  if (_cast && this->dataPtr->light->getType() !=
+      Ogre::Light::LT_DIRECTIONAL)
   {
-    this->dataPtr->light->setCastShadows(false);
+    if (this->dataPtr->shadowCameraSetup.isNull())
+    {
+      auto *setup = new Ogre::FocusedShadowCameraSetup();
+      this->dataPtr->shadowCameraSetup = Ogre::ShadowCameraSetupPtr(setup);
```

Comment:
nit: consider using `std::make_shared` to combine these two lines if the type conversions permit it

Reply:
`Ogre::ShadowCameraSetupPtr` is an ogre shared pointer but not a `std::shared_pointer` so I was not able to use `make_shared`. I've combined the two lines nevertheless.